tools/mine-localisation-files.py

```diff
@@ -0,0 +1,208 @@
+    best = ''
+    weight = 0
+    for possibility in possibilities:
+        for check in checks:
+            if check[0] > weight and check[1](possibility):
+                best = possibility
+                weight = check[0]
+
+    if not best:
+        # Just return the first one in the list
+        best = possibilities[0]
+
+    return best
+
+def convert_file(filename, expected, destination):
+    print('Copying and converting file: {}'.format(filename))
+
+    path = get_path(filename, expected)
+    print('Found version at: {}'.format(path))
+    if path:
+        final_path = destination + '/' + filename
+        print('Copying file to: {}'.format(final_path))
+        shutil. copyfile(path, final_path)
+        convert_firefox_to_browser(final_path)
+
+def mine_locale(locale, date):
+    repo_local = '/tmp/l10n-repo'
+    repo_remote = 'https://hg.mozilla.org/l10n-central/' + locale
+    retcode = 0
+    cloned = False
+
+    # Perform checks
+    git_root = get_git_root()
+    if not git_root:
+        print('You must execute this inside the embedlite-components git tree')
+        retcode = 128
+
+    if retcode == 0:
+        print('Git root: {}'.format(git_root))
+
+        cwd_orig = os.getcwd()
+
+        #os.chdir('/tmp')
+
+        # Clone the repository
+        print('Cloning repository {} to {}'.format(repo_remote, repo_local))
+        retcode = subprocess.call(['hg', 'clone', repo_remote, repo_local, '-U'])
+        if retcode == 0:
+            print('Successfully cloned')
+            cloned = True
+        else:
+            print('Error cloning')
+
+    if retcode == 0:
+        os.chdir(repo_local)
+
+        # Search for the appropriate commit
+        print('Searching for an appropriate commit')
+        checkdate = date
+        revision = ''
+        while revision == '':
+            datestring = date.isoformat()
+            revision = subprocess.check_output(['hg', 'log', '--date', datestring, '--limit', '1', '--template', '"{node}"'], encoding='ascii').strip()
+            date += datetime.timedelta(days=-1)
+        print("Found revision {} on {}".format(revision, datestring))
+
+        # Update to the revision
+        print('Checking out revision {}'.format(revision))
+        retcode = subprocess.call(['hg', 'update', '--rev', revision])
+        if retcode == 0:
+            print('Revision Successfully set')
+        else:
+            print('Error updating to revision')
+
+    if retcode == 0:
+        folder = git_root + '/overrides/' + locale
+        try:
+            os.mkdir(folder)
+        except:
+            # Just skip it if it already exists
+            pass
+
+        # Search for the right files
+        # Copy the files
+        # Transform the files
+
+        print('Duplicating non-localised files')
+
+        # overrides/en-US/brand.dtd
+        shutil. copyfile(git_root + '/overrides/en-US/brand.dtd', folder + '/brand.dtd')
```

Comment:
whitespace after shutil. ?

Reply:
I'm a bit surprised python wasn't more unhappy about this. Thanks for spotting it, I've removed the space.